src/dataset_exec.rs

```diff
@@ -31,12 +31,12 @@ use datafusion::arrow::error::Result as ArrowResult;
 use datafusion::arrow::record_batch::RecordBatch;
 use datafusion::error::{DataFusionError as InnerDataFusionError, Result as DFResult};
 use datafusion::execution::context::TaskContext;
-use datafusion::logical_plan::{combine_filters, Expr};
 use datafusion::physical_expr::PhysicalSortExpr;
 use datafusion::physical_plan::stream::RecordBatchStreamAdapter;
 use datafusion::physical_plan::{
     DisplayFormatType, ExecutionPlan, Partitioning, SendableRecordBatchStream, Statistics,
 };
+use datafusion_expr::{combine_filters, Expr};
```

Comment:
I think `datafusion::logical_plan` in [expression.rs#L22](https://github.com/apache/arrow-datafusion-python/blob/master/src/expression.rs#L22)  needs to be replaced with `datafusion_expr`